### PR TITLE
[tensorflow] [build] [test] TF binaries for 2.1.2 with CUDA10.1

### PR DESCRIPTION
**Pull request by Teng-xu**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saimidu — COMMENTED** [review]
Please upgrade `sagemaker<2` to `sagemaker>=2,<3` in all dockerfiles.

File: tensorflow/training/docker/2.1/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -27,7 +27,7 @@ ARG PIP=pip3
 
 ARG OPENSSL_VERSION=1.1.1g
 
-ARG TF_URL=https://tensorflow-aws.s3-us-west-2.amazonaws.com/2.1.1/AmazonLinux/cpu/final/tensorflow-2.1.1-cp36-cp36m-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.1_aws/20201030-193011/cpu/tensorflow_cpu-2.1.2-cp36-cp36m-manylinux2010_x86_64.whl
 
 # The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
```
[the remark this answers is not in the document]
Comment (Teng-xu) [reply tensorflow/training/docker/2.1/py3/Dockerfile.cpu:33]: Done

File: tensorflow/training/docker/2.1/py3/cu101/Dockerfile.gpu [hunk 2]
```diff
@@ -27,7 +27,7 @@ ARG PIP=pip3
 
 ARG OPENSSL_VERSION=1.1.1g
 
-ARG TF_URL=https://tensorflow-aws.s3-us-west-2.amazonaws.com/2.1.1/AmazonLinux/gpu/final/tensorflow_gpu-2.1.1-cp36-cp36m-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.1_aws/20201030-193011/gpu/tensorflow_gpu-2.1.2-cp36-cp36m-manylinux2010_x86_64.whl
 
 # The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
```
[the remark this answers is not in the document]
Comment (Teng-xu) [reply tensorflow/training/docker/2.1/py3/cu101/Dockerfile.gpu:33]: Done

**Review by saimidu — COMMENTED** [review]
@Teng-xu LGTM. Let's see how the build goes. Thanks for the changes

**Review by saimidu — COMMENTED** [review]
LGTM. I can approve. Could you revert the change to `tensorflow/buildspec.yml` so that it builds TF 2.3.1 again?

**Review by saimidu — APPROVED** [review]

**Pull request merged**